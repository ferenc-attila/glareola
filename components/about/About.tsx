import { Text, View } from "react-native";

export default function About() {
    return (
      <>
        <View>
            <Text>
                {'Read rarity observations from birding.hu'}
            </Text>
        </View>
      </>
    );
}
