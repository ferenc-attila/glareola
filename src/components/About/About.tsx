import { ScrollView, StyleSheet, Text, View } from 'react-native';
import { ABOUT } from '../../content';

const createTextComponents = (text: string, index: number | string, style: object) => {
            return <Text
                key={index}
                style={style}
            >
                {text}
            </Text>
}

export const About = () => {
    return (
        <View style={styles.textContainer}>
            <ScrollView>
                <View style={styles.textContainer}>
                    <Text style={styles.title}>
                        {ABOUT.APP.TITLE}
                    </Text>
                    {Object.values(ABOUT.APP).map((value, index) => {
                        if (typeof value === 'string' && value !== ABOUT.APP.TITLE) {
                            return createTextComponents(value, index, styles.description);
                            }
                        }
                    )}
                    <Text style={styles.description}>
                        {ABOUT.APP.LICENSE.SHORT_NAME}
                    </Text>
                    <Text style={styles.title}>
                        {ABOUT.COVER_IMAGE.TITLE}
                    </Text>
                    {Object.values(ABOUT.COVER_IMAGE).map((value, index) => {
                        if (typeof value === 'string' && value !== ABOUT.COVER_IMAGE.TITLE) {
                            return createTextComponents(value, index, styles.description);
                            }
                        }
                    )}
                    <Text style={styles.description}>
                        {ABOUT.COVER_IMAGE.LICENSE.SHORT_NAME}
                    </Text>
                    <Text style={styles.title}>
                        {ABOUT.ICON_IMAGE.TITLE}
                    </Text>
                    {Object.values(ABOUT.ICON_IMAGE).map((value, index) => {
                        if (typeof value === 'string' && value !== ABOUT.ICON_IMAGE.TITLE) {
                            return createTextComponents(value, index, styles.description);
                            }
                        }
                    )}
                    <Text style={styles.description}>
                        {ABOUT.ICON_IMAGE.LICENSE.SHORT_NAME}
                    </Text>
                </View>
            </ScrollView>
        </View>
    );
}

const styles = StyleSheet.create({
    textContainer: {
        flex: 1,
        backgroundColor: '#2e3a24',
        alignItems: 'center',
    },
    title: {
        marginTop: 25,
        color: '#fff',
        fontWeight: '900',
        fontSize: 22,
        padding: 10,
    },
    description: {
        color: '#fff',
        fontSize: 16,
        padding: 5,
        textAlign: 'center',
    },
});
