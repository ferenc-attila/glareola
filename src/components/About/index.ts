export { About } from './About';
export { ABOUT } from './constants';
