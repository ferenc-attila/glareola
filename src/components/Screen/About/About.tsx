import { ScrollView, StyleSheet, Text, View } from 'react-native';

import { ABOUT } from '../../../constants';
import i18n from '../../../utils/localization/localization';

export const About = () => {
    return (
        <View style={styles.textContainer}>
            <ScrollView>
                <View style={styles.textContainer}>
                    <Text style={styles.title}>{i18n.t('APP_NAME')}</Text>
                    <Text style={styles.description}>{i18n.t('APP_DESCRIPTION')}</Text>
                    <Text style={styles.description}>{i18n.t('APP_AUTHOR')}</Text>
                    <Text style={styles.description}>{ABOUT.APP.VERSION}</Text>
                    <Text style={styles.description}>{ABOUT.APP.DATE}</Text>
                    <Text style={styles.description}>{ABOUT.APP.LICENSE.SHORT_NAME}</Text>
                    <Text style={styles.title}>{i18n.t('COVER_IMAGE_NAME')}</Text>
                    <Text style={styles.description}>{i18n.t('COVER_IMAGE_DESCRIPTION')}</Text>
                    <Text style={styles.description}>{i18n.t('COVER_IMAGE_AUTHOR')}</Text>
                    <Text style={styles.description}>{i18n.t('COVER_IMAGE_DATE')}</Text>
                    <Text style={styles.description}>{i18n.t('COVER_IMAGE_LOCATION')}</Text>
                    <Text style={styles.description}>{ABOUT.COVER_IMAGE.LICENSE.SHORT_NAME}</Text>
                    <Text style={styles.title}>{i18n.t('ICON_IMAGE_NAME')}</Text>
                    <Text style={styles.description}>{i18n.t('ICON_IMAGE_DESCRIPTION')}</Text>
                    <Text style={styles.description}>{i18n.t('ICON_IMAGE_AUTHOR')}</Text>
                    <Text style={styles.description}>{i18n.t('ICON_IMAGE_DATE')}</Text>
                    <Text style={styles.description}>{i18n.t('ICON_IMAGE_LOCATION')}</Text>
                    <Text style={styles.description}>{ABOUT.ICON_IMAGE.LICENSE.SHORT_NAME}</Text>
                </View>
            </ScrollView>
        </View>
    );
};

const styles = StyleSheet.create({
    textContainer: {
        flex: 1,
        backgroundColor: '#2e3a24',
        alignItems: 'center',
    },
    title: {
        marginTop: 25,
        color: '#fff',
        fontWeight: '900',
        fontSize: 22,
        padding: 10,
    },
    description: {
        color: '#fff',
        fontSize: 16,
        padding: 5,
        textAlign: 'center',
    },
});
