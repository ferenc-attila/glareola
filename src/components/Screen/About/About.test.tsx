import { render } from '@testing-library/react-native/build/pure';

import { About } from './About';

describe('About', () => {
    test('should render correctly', () => {
        const container = render(<About />);
        expect(container.toJSON()).toMatchSnapshot();
    });
});
