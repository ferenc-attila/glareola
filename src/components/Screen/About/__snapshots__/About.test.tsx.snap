// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About should render correctly 1`] = `
<RCTScrollView>
  <View>
    <Text
      style={
        {
          "color": "#fff",
          "fontSize": 18,
          "fontWeight": "700",
          "padding": 10,
          "textAlign": "center",
          "textAlignVertical": "bottom",
        }
      }
    >
      Glareola
    </Text>
    <Text
      style={
        {
          "color": "#fff",
          "fontSize": 14,
          "fontWeight": "500",
        }
      }
    >
      A crawler for rarity observations
    </Text>
    <Text
      style={
        {
          "color": "#fff",
          "fontSize": 14,
          "fontWeight": "500",
        }
      }
    >
      Attila Ferenc
    </Text>
    <Text
      style={
        {
          "color": "#fff",
          "fontSize": 14,
          "fontWeight": "500",
        }
      }
    >
      0.0.8
    </Text>
    <Text
      style={
        {
          "color": "#fff",
          "fontSize": 14,
          "fontWeight": "500",
        }
      }
    >
      2023-10-10
    </Text>
    <Text
      style={
        {
          "color": "#fff",
          "fontSize": 14,
          "fontWeight": "500",
        }
      }
    >
      MIT
    </Text>
    <Text
      style={
        {
          "color": "#fff",
          "fontSize": 18,
          "fontWeight": "700",
          "padding": 10,
          "textAlign": "center",
          "textAlignVertical": "bottom",
        }
      }
    >
      Cover Image
    </Text>
    <Text
      style={
        {
          "color": "#fff",
          "fontSize": 14,
          "fontWeight": "500",
        }
      }
    >
      Collared pratincole (Glareola pratincola)
    </Text>
    <Text
      style={
        {
          "color": "#fff",
          "fontSize": 14,
          "fontWeight": "500",
        }
      }
    >
      Saswat Mishra
    </Text>
    <Text
      style={
        {
          "color": "#fff",
          "fontSize": 14,
          "fontWeight": "500",
        }
      }
    >
      2018-03-30
    </Text>
    <Text
      style={
        {
          "color": "#fff",
          "fontSize": 14,
          "fontWeight": "500",
        }
      }
    >
      Nal Sarovar, Gujarat, India
    </Text>
    <Text
      style={
        {
          "color": "#fff",
          "fontSize": 14,
          "fontWeight": "500",
        }
      }
    >
      CC-BY-SA-4.0
    </Text>
    <Text
      style={
        {
          "color": "#fff",
          "fontSize": 18,
          "fontWeight": "700",
          "padding": 10,
          "textAlign": "center",
          "textAlignVertical": "bottom",
        }
      }
    >
      Icon Image
    </Text>
    <Text
      style={
        {
          "color": "#fff",
          "fontSize": 14,
          "fontWeight": "500",
        }
      }
    >
      Collared pratincole (Glareola pratincola) portrait
    </Text>
    <Text
      style={
        {
          "color": "#fff",
          "fontSize": 14,
          "fontWeight": "500",
        }
      }
    >
      Ludovic Hirlimann
    </Text>
    <Text
      style={
        {
          "color": "#fff",
          "fontSize": 14,
          "fontWeight": "500",
        }
      }
    >
      2010-05-16
    </Text>
    <Text
      style={
        {
          "color": "#fff",
          "fontSize": 14,
          "fontWeight": "500",
        }
      }
    >
      Amboseli National Park, Kenya
    </Text>
    <Text
      style={
        {
          "color": "#fff",
          "fontSize": 14,
          "fontWeight": "500",
        }
      }
    >
      CC-BY-SA-3.0
    </Text>
  </View>
</RCTScrollView>
`;
